src/main/java/uk/gov/ons/validation/service/InvokeConfig.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.ons.validation.service;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import uk.gov.ons.validation.entity.QuestionInputData;
+
+import static java.lang.String.format;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Log4j2
+public class InvokeConfig {
+
+    private String finalQuestCode;
+    private String finalDerivedQuestCode;
+    private String finalQuestCodeValue;
+    private String finalDerivedQuestValue;
+    private boolean isQuestionCodeFound;
```

Comment:
I don't think you need these extra boolean attributes to complicate things. You could have one boolean method which returns true if both the question and derived question information is found.

Reply:
Extra Boolean will give detail explanation whether Question Match found and in a similar way Derived Question Found. I added another method isBothFound() which verifies both Question and Derived question found.